src/components/TimelineRender/hook/useOnClickNewMsg.ts

```diff
@@ -0,0 +1,38 @@
+import { queryCache } from 'core/cache/query';
+import { mergeAndSortUniqueDbEvents } from 'utils/common';
+import { useQueryCacheId } from './useQueryCacheId';
+import { Dispatch, SetStateAction, useCallback } from 'react';
+import { CallWorker } from 'core/worker/caller';
+import { Filter } from 'core/nostr/type';
+import { Event } from 'core/nostr/Event';
+import { DbEvent } from 'core/db/schema';
+
+export interface ClickNewMsgProp {
+  queryCacheId: string;
+  setMsgList: Dispatch<SetStateAction<DbEvent[]>>;
+  setNewComingMsg: Dispatch<SetStateAction<DbEvent[]>>;
```

Comment:
Passing the setStateAction of other states to be called elsewhere is not a good approach, as it can easily lead to confusion in data flow. Perhaps a better implementation should be considered.